### PR TITLE
Distinguish between Forth and Fortran filetypes for *.f files

### PR DESCRIPTION
**Pull request by dkearns**
Also adds *.4th as a Forth filetype

@jkotlinski, could you please run your eyes over this?

### CODE REVIEW

**Comment by jkotlinski** [comment]
Hey, sorry, I missed the notification. I will have a look.

File: runtime/autoload/dist/ft.vim [hunk 1]
```diff
@@ -286,6 +286,39 @@ export def FTe()
...
+    exe "setf " .. g:filetype_f
+    return
+  endif
+
+  var first_line = getline(nextnonblank(1))
+
+  # SwiftForth block comment, line is usually filled with '-' or '='
+  var header_comment   = first_line =~ '^{\%(\s\|$\)'
+  # SwiftForth OPTIONAL, sometimes appears before the header comment
+  var optional_include = first_line =~ '^OPTIONAL\s'
```
Comment (jkotlinski) [inline runtime/autoload/dist/ft.vim]: Could the SwiftForth checks give false positives for F#? If not, I think they would be better moved to the IsForth() function.

**Comment by dkearns** [comment]
I don't believe so.  I'll push the extra tests into IsForth().

Thanks.

File: runtime/filetype.vim [hunk 2]
```diff
@@ -710,16 +710,19 @@ au BufNewFile,BufRead auto.master		setf conf
...
-au BufNewFile,BufRead *.ft,*.fth		setf forth
+au BufNewFile,BufRead *.ft,*.fth,*.4th		setf forth
 
 " Reva Forth
 au BufNewFile,BufRead *.frt			setf reva
 
 " Fortran
 if has("fname_case")
-  au BufNewFile,BufRead *.F,*.FOR,*.FPP,*.FTN,*.F77,*.F90,*.F95,*.F03,*.F08	 setf fortran
+  au BufNewFile,BufRead *.F,*.FOR,*.FPP,*.FTN,*.F77,*.F90,*.F95,*.F03,*.F08	setf fortran
```
Comment (jkotlinski) [inline runtime/filetype.vim:734]: Is it intentional that *.F is still mapped to Fortran only?

Comment (dkearns) [reply runtime/filetype.vim:734]: It is on the grounds that:

- Fortran has some conventions for uppercase extensions that makes them meaningful.
- We don't support uppercase for any other Forth extensions or other filetypes in general.
- SwiftForth (a notable user of `*.f`) doesn't implicitly recognise `*.F` as it does `*f`.

Is there some history of intentionally capitalised extensions?

Comment (jkotlinski) [reply runtime/filetype.vim:734]: Not that I am aware of. I was simply unaware of Fortran conventions.

**Comment by yegappan** [comment]
Can you please update and rebase this PR to the latest?

**Pull request closed without merging**